plugins/modules/win_acl.ps1

```diff
@@ -103,6 +104,14 @@ if ($path_qualifier -eq "HKCC:" -and (-not (Test-Path -LiteralPath HKCC:\))) {
     New-PSDrive -Name HKCC -PSProvider Registry -Root HKEY_CURRENT_CONFIG > $null
 }
 
+$path = Resolve-Path -LiteralPath $path
+Do {
+    $PathDetails = Get-ItemProperty -LiteralPath $path
+    if ($PathDetails.Target -and $PathDetails.Target.GetType().Name -eq "String[]" -and $follow_links) {
```

Comment:
I believe `Target` is a property added in WinPS 5.1 but we still need to support pwsh 3 and 4. We have some module utils that can help with links.

```powershell
# Put at the top of the script with #AnsibleRequires to reference this module util
#Requires -Module Ansible.ModuleUtils.LinkUtil

...

$follow_links = $true

while ($follow_links) {
    try {
        $link_info = Get-Link -Path $path
    } catch {
        $link_info = $null
    }

    if ($link_info -and $link_info.Type -in @("SymbolicLink", "JunctionPoint")) {
        $path = $link_info.AbsolutePath
    }
    else {
        break
    }
}
```

Reply:
Sorry but I don't understand the logic of your loop.
You never update $follow_links inside the loop so it should be an infinite loop no ?
My initial idea was to walk over symlinks/junctions.

Let's imagine a Symlink pointing on a Junction wich is a valid use case.

Reply:
The `while ($follow_links)` will ensure the code is run if `$follow_links` is `$true`. It will continuously loop through the code until either `Get-Link` fails or there is no link target. This is done through the if check at the end where `break` is called to exit the loop if either `$link_info` is not set (catch block or the path is not a link) or the link type is not a `SymbolicLink` or `JunctionPoint`. This ensures that it will continue to follow the link's targets until that condition is met.